rtl/eclrtl/eclrtl.cpp

```diff
@@ -3520,8 +3538,26 @@ hash64_t rtlHash64VUnicode(UChar const * k, hash64_t initval)
 unsigned rtlHash32Data(size32_t len, const void *buf, unsigned hval)
 {
     const unsigned char *bp = (const unsigned char *)buf;   /* start of buffer */
-    const unsigned char *be = bp + len;     /* beyond end of buffer */
 
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+    //This possibly breaks the aliasing rules for c++, but I can't see it causing any problems
+    while (len >= sizeof(unsigned))
```

Comment:
It helps if it is a multiple of 8, but would slow it down if it wasn't.  This is used on string fields, so I'm not convinced...